HumanNeverDie/Projects/Features/HistoryFeature/Sources/Calendar/ViewModel/AMDCalendarViewModel.swift

```diff
@@ -0,0 +1,207 @@
+//
+//  AMDCalendarViewModel.swift
+//  HistoryFeature
+//
+//  Created by Seulki Lee on 6/25/25.
+//
+
+import SwiftUI
+import DesignSystem
+
+public struct SugarIntakeRecord {
+  let date: Date
+  let value: Int
+}
+
+public struct WeekdayValue: Identifiable {
+  public let id = UUID()
+  let label: String
+  let color: Color
+}
+
+public struct DateValue: Identifiable {
+  public var id: UUID = UUID()
+  var day: Int
+  var date : Date
+}
+
+final class AMDCalendarViewModel: ObservableObject {
+  @Published var currentMonth: Int = 0
```

Comment:
ViewModel은 `ViewModelable`을 채택하여 단방향 데이터 흐름(UDF)을 가져가도록 하는 것으로 알고 있는데요, 

`currentMonth`처럼 View에서 접근 가능한 프로퍼티인 경우 이 흐름이 지켜지지 않게 됩니다.

`ViewModelable` 프로토콜을 채택하여 현재 구조를 개선하는 걸 추천드리고, 아니라면 최소한 아래처럼 직접 접근하여 수정할 수 없도록 개선되어야할 것 같아요~

```suggestion
final class AMDCalendarViewModel: ObservableObject {
  @Published private(set) var currentMonth: Int = 0
```

Reply:
ViewModelable은 제가 아직 잘 알지 못했던 부분이라, 말씀 주신 내용을 바탕으로 개념부터 다시 공부해보고 구조를 개선해보겠습니다! 

감사합니다!